platform/admin/admin-configuration-configinstaller/src/test/java/org/codice/ddf/admin/configuration/ConfigurationInstallerTest.java

```diff
@@ -0,0 +1,175 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.admin.configuration;
+
+import static org.hamcrest.Matchers.empty;
+import static org.hamcrest.Matchers.hasSize;
+import static org.hamcrest.core.Is.is;
+import static org.junit.Assert.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+import static org.osgi.service.cm.ConfigurationEvent.CM_DELETED;
+import static org.osgi.service.cm.ConfigurationEvent.CM_UPDATED;
+
+import java.io.File;
+import java.util.Dictionary;
+import java.util.Hashtable;
+import java.util.Map;
+
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+import org.osgi.service.cm.Configuration;
+import org.osgi.service.cm.ConfigurationAdmin;
+import org.osgi.service.cm.ConfigurationEvent;
+
+@RunWith(MockitoJUnitRunner.class)
+public class ConfigurationInstallerTest {
+    private static final String FELIX_FILENAME_PROP = "felix.fileinstall.filename";
+
+    // Will be tracked in the installer's map after init()
+    private static final String PID_001 = "001";
+
+    // Will NOT be tracked in the installer's map after init()
+    private static final String PID_002 = "002";
+
+    @Mock
+    private Configuration configuration;
+
+    @Mock
+    private ConfigurationAdmin configurationAdmin;
+
+    @Mock
+    private ConfigurationEvent configurationEvent;
+
+    private ConfigurationInstaller installer;
+
+    @Rule
+    public TemporaryFolder temporaryFolder = new TemporaryFolder();
+
+    private File file1a;
+
+    private File file1b;
+
+    @Before
+    public void before() throws Exception {
+        temporaryFolder.create();
+        when(configurationEvent.getType()).thenReturn(CM_UPDATED);
+
+        file1a = temporaryFolder.newFile();
+        file1b = temporaryFolder.newFile();
+
+        // Ensure init takes the last value for a particular key
+        Configuration mockConfig1a = createMockConfig(PID_001, file1a.toURI());
```

Comment:
I don't think that's necessary as that code is unit tested elsewhere. See `FelixConfigTest`.